src/core/js/router.js

```diff
@@ -17,12 +16,57 @@ define(function(require) {
                 document.title = Adapt.course.get('title');
             });
             this.listenTo(Adapt, 'navigation:backButton', this.navigateToPreviousRoute);
+            this.listenTo(Adapt, "router:navigateTo", this.handleRoute);
         },
 
         routes: {
-            "":"handleCourse",
-            "id/:id":"handleId",
-            ":pluginName(/*location)(/*action)": "handlePluginRouter"
+            "":"handleRoute",
+            "id/:id":"handleRoute",
+            ":pluginName(/*location)(/*action)": "handleRoute"
+        },
+
+        handleRoute: function() {
+            var args = [].slice.call(arguments, 0, arguments.length);
+            if (arguments[arguments.length-1] === null) args.pop();
+
+            
+            if (!this._isCircularNavigationInProgress) {
```

Comment:
an explanatory comment for the `_isCircularNavigationInProgress` property would be useful